cirq-core/cirq/testing/consistent_unitary.py

```diff
@@ -0,0 +1,68 @@
+# Copyright 2023 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from typing import Any
+import cirq
+import numpy as np
+
+
+def assert_unitary_is_consistent(val: Any, ignoring_global_phase: bool = False):
+    if not cirq.has_unitary(val):
+        return
```

Comment:
return if `val` is not `isinstance(cirq.Operation, cirq.Gate)` since the check if mainly for these two types?